src/modules/dashboard/sagas/actions.ts

```diff
@@ -1086,6 +1086,180 @@ function* editColonyAction({
   return null;
 }
 
+function* createSetUserRolesAction({
+  payload: {
+    colonyAddress,
+    domainId,
+    userAddress,
+    roles,
+    colonyName,
+    annotationMessage,
+  },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_ACTION_USER_ROLES_SET>) {
+  let txChannel;
+  try {
+    const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+
+    if (!userAddress) {
+      throw new Error('User address not set for setUserRole transaction');
+    }
+
+    if (!domainId) {
+      throw new Error('Domain id not set for setUserRole transaction');
+    }
+
+    if (!roles) {
+      throw new Error('Roles not set for setUserRole transaction');
+    }
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    const batchKey = 'setUserRoles';
+
+    let annotationMessageIpfsHash = null;
+    if (annotationMessage) {
+      annotationMessageIpfsHash = yield call(
+        ipfsUpload,
+        JSON.stringify({
+          annotationMessage,
+        }),
+      );
+    }
```

Comment:
Those sagas was refactored in @rdig https://github.com/JoinColony/colonyDapp/pull/2359
We need to adjust to those change. You would need to cherry-pick this commit->
https://github.com/JoinColony/colonyDapp/pull/2359/commits/d32a4a67748d09203cef281539718bf03bdc5313

Reply:
:point_up: What @alicjakujawa said :)

Reply:
This wasn't changed yet...